applications/ShallowWaterApplication/custom_utilities/shallow_water_utilities.cpp

```diff
@@ -111,4 +111,52 @@ void ShallowWaterUtilities::IdentifySolidBoundary(ModelPart& rSkinModelPart, dou
     }
 }
 
+void ShallowWaterUtilities::IdentifyWetDomain(ModelPart& rModelPart, Flags WetFlag, double Thickness)
+{
+    #pragma omp parallel for
+    for (int i = 0; i < static_cast<int>(rModelPart.NumberOfNodes()); ++i)
+    {
+        auto it_node = rModelPart.NodesBegin() + i;
+        const double height = it_node->FastGetSolutionStepValue(HEIGHT);
+        it_node->Set(WetFlag, (height > Thickness));
+    }
+
+    #pragma omp parallel for
+    for (int i = 0; i < static_cast<int>(rModelPart.NumberOfElements()); ++i)
+    {
+        auto it_elem = rModelPart.ElementsBegin() + i;
+
+        bool wet_element = false;
+        for(auto& node : it_elem->GetGeometry())
+        {
+            if (node.Is(WetFlag)) {
+                wet_element = true;  // It means there is almost a wet node
+                break;
+            }
+        }
+
+        it_elem->Set(WetFlag, wet_element);
+    }
+}
+
+void ShallowWaterUtilities::ComputeVisualizationWaterHeight(ModelPart& rModelPart, Flags WetFlag, double SeaWaterLevel)
+{
+    #pragma omp parallel for
+    for (int i = 0; i < static_cast<int>(rModelPart.NumberOfNodes()); ++i)
+    {
+        auto it_node = rModelPart.NodesBegin() + i;
+        if (it_node->Is(WetFlag)) {
+            if (it_node->FastGetSolutionStepValue(TOPOGRAPHY) > SeaWaterLevel) {
+                it_node->SetValue(WATER_HEIGHT, it_node->FastGetSolutionStepValue(HEIGHT));
+            }
+            else {
+                it_node->SetValue(WATER_HEIGHT, it_node->FastGetSolutionStepValue(FREE_SURFACE_ELEVATION) - SeaWaterLevel);
+            }
+        }
+        else {
+            it_node->SetValue(WATER_HEIGHT, GID_NO_RESULT_DOUBLE);
```

Comment:
Are you doing this only for visualization purpose? If that's the case, I was wondering if there is a cleaner way to do this...

Reply:
This is for visualization purpose, but it makes physical sense. The water height is undefined on the dry domain. This value means undefined for GiD, but I'm wondering if there is a more generalized undefined value.
Zero and negative values make physical sense. This value becomes negative at the sine of a wave.

Reply:
I'm going to try with `std::lowest()`

Reply:
Great!
If there is  no other way, it's ok for me, though

Reply:
I have substituted `GID_NO_RESULT_DOUBLE` by `std::numeric_limits<float>::lower()`

I'm using `float`  instead of `double` because `gidpost` library checks this exact value, not a lower value